### ISSUE TITLE
Unloaded projects break Render Template

### ISSUE THREAD

**Issue by wonea** [frhagn/Typewriter#168]
In Visual Studio 2017 clicking Render Template gives error message. Saving individual files still generates TypeScript files.

![image](https://cloud.githubusercontent.com/assets/557192/23857282/ab0d3f46-07f4-11e7-857f-cb38c31480fe.png)

![image](https://cloud.githubusercontent.com/assets/557192/23857319/cdcbbe04-07f4-11e7-8817-24b0c88be3cf.png)

TypeWriter 1.10, Visual Studio 15.0.0-RTW+26228.4

**Comment by frhagn** [comment]
Hi @wonea,

I can't reproduce this error on any of my machines, but I've just published a new version (1.11.0) which is built using VS 2017 RTM.. It would be great if you could try out the new version to see if it resolves this issue.

**Comment by wonea** [comment]
Thanks for the new build! It's work for one project, but not another. I've got a feeling it maybe my project is breaking TypeWriter somehow. The TypeWriter out is not showing;

![image](https://cloud.githubusercontent.com/assets/557192/23898187/0a30a89c-08a8-11e7-829d-0e12bc559ca6.png)

Can't seem to find any VS 2017 log files.

**Comment by wonea** [comment]
My earlier message was a red herring, as the TypeWriter output only appears once it's been used.

However I've noticed in the project I'm having troubles with I get the following warning;

14:56:47.624 WARNING: Falied to save source map for 'C:\Development\My.Core.DTO\Process\Process.cs'. Project type not supported.
15:19:42.641 WARNING: Falied to save source map for 'C:\Development\My.Core.DTO\Process\Process.cs'. Project type not supported.

**Comment by wonea** [comment]
I'm interesting in looking into this warning. I've included an example project showing the warning.

![image](https://cloud.githubusercontent.com/assets/557192/23944048/5f806dde-0969-11e7-9516-806f3bec0aa5.png)

[WebSocket.Common (bug report).zip](https://github.com/frhagn/Typewriter/files/844080/WebSocket.Common.bug.report.zip)

**Comment by aschwartz-satcom** [comment]
I was able to dig into my own encounter with this same exception being thrown on "Render Template" context menu action.  I grabbed the Typewriter source and compiled the extension myself (used jotabe-net's fork since it looked like that one resolves some missing dependencies). 
 
Once I got the debugger attached to an "Experimental Instance" of VS with the extension loaded, I opened up my problematic Solution and clicked the "Render Template" context menu action on one of my templates (all of them were failing with the same E_NOTIMPL exception).

To my surprise, the debugger revealed that it was choking on my [unloaded](https://msdn.microsoft.com/en-us/library/tt479x1t(v=vs.100).aspx) "Tests" project.  I guess unloaded projects throw a Not Implemented Exception when you try to get their FullName:

![screen shot 2017-05-01 at 4 08 32 pm](https://cloud.githubusercontent.com/assets/15252204/25593130/5e446908-2e89-11e7-8344-5760b43fc36b.png)
![screen shot 2017-05-01 at 4 07 17 pm](https://cloud.githubusercontent.com/assets/15252204/25593131/5e4a38e2-2e89-11e7-810e-7114aebe48a2.png)

Reloading the project works around the issue.

**Comment by wonea** [comment]
@aschwartz-satcom Thanks for finding this out. @frhagn I can reproduce this bug unloading a project within the solution.

**Comment by frhagn** [comment]
Awesome @aschwartz-satcom! Working on a fix..

**Comment by frhagn** [comment]
Fixed in 1.12.0

**Comment by JavanXD** [comment]
I am getting these outputs all the time:
```
13:28:09.443 WARNING: Falied to save source map for 'C:\Projects\asd\src\DataBaseService\Models\ClassificationEnum.cs'. Project type not supported.
13:28:09.445 WARNING: Falied to save source map for 'C:\Projects\asd\src\DataBaseService\Models\ContactTypeEnum.cs'. Project type not supported.
```
VS 2017 15.3.4
Typewriter 1.12.2

How can i disable these warnings?
